core/services/chainlink/config.go

```diff
@@ -49,42 +51,139 @@ type Config struct {
 
 // ValidateConfig returns an error if the Config is not valid for use, as-is.
-func (c *RawConfig) ValidateConfig() (err error) {
-	if v, ok := (*c)["Enabled"]; ok {
+func (c RawConfig) ValidateConfig() (err error) {
+	if v, ok := c["Enabled"]; ok {
 		if _, ok := v.(bool); !ok {
 			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "Enabled", Value: v, Msg: "expected bool"})
 		}
 	}
-	if v, ok := (*c)["ChainID"]; ok {
-		if _, ok := v.(string); !ok {
-			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "ChainID", Value: v, Msg: "expected string"})
+	chainID, exists := c["ChainID"]
+	if !exists {
+		err = multierr.Append(err, commonconfig.ErrMissing{Name: "ChainID", Msg: "required for all chains"})
+	} else {
+		chainIDStr, ok := chainID.(string)
+		if !ok {
+			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "ChainID", Value: chainID, Msg: "expected string"})
+		} else if chainIDStr == "" {
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "ChainID", Msg: "required for all chains"})
+		}
+	}
+	nodes, exists := c["Nodes"]
+	if !exists {
+		err = multierr.Append(err, commonconfig.ErrMissing{Name: "Nodes", Msg: "expected at least one node"})
+	} else {
+		nodeMaps, ok := nodes.([]any)
+		if !ok {
+			err = multierr.Append(err, commonconfig.ErrInvalid{Name: "Nodes", Value: nodes, Msg: "expected array of node configs"})
+		} else if len(nodeMaps) == 0 {
+			err = multierr.Append(err, commonconfig.ErrEmpty{Name: "Nodes", Msg: "expected at least one node"})
+		} else {
+			for i, node := range nodeMaps {
+				nodeConfig, ok := node.(map[string]any)
+				if !ok {
+					err = multierr.Append(err, commonconfig.ErrInvalid{Name: fmt.Sprintf("Nodes.%d", i), Value: nodeConfig, Msg: "expected node config map"})
+				} else {
+					nodeName, exists := nodeConfig["Name"]
+					if !exists {
+						err = multierr.Append(err, commonconfig.ErrMissing{Name: fmt.Sprintf("Nodes.%d.Name", i), Msg: "required for all nodes"})
+					} else {
+						nodeNameStr, ok := nodeName.(string)
+						if !ok {
+							err = multierr.Append(err, commonconfig.ErrInvalid{Name: fmt.Sprintf("Nodes.%d.Name", i), Value: nodeName, Msg: "expected string"})
+						} else if nodeNameStr == "" {
+							err = multierr.Append(err, commonconfig.ErrEmpty{Name: fmt.Sprintf("Nodes.%d.Name", i), Msg: "required for all nodes"})
+						}
+					}
+				}
+			}
 		}
 	}
 	return err
 }
 
-func (c *RawConfig) IsEnabled() bool {
-	if c == nil {
-		return false
-	}
-
-	enabled, ok := (*c)["Enabled"].(bool)
+func (c RawConfig) IsEnabled() bool {
+	enabled, ok := c["Enabled"].(bool)
 	return ok && enabled
 }
 
-func (c *RawConfig) ChainID() string {
-	if c == nil {
-		return ""
-	}
-
-	chainID, _ := (*c)["ChainID"].(string)
+func (c RawConfig) ChainID() string {
+	chainID, _ := c["ChainID"].(string)
 	return chainID
 }
 
+func (c RawConfig) SetFrom(config RawConfig) error {
+	return mergo.Merge(&c, config, mergo.WithOverride, mergo.WithAppendSlice)
```

Comment:
Is this safe to do? `WithAppendSlice` sounds correct for our Chains, but it would not be correct for any other fields that I can think of.

Reply:
Oh actually, it is not correct for chains either, because it would append matching chain IDs (and thus duplicate) instead of merging them, right?

Reply:
Ah but the chain logic is higher level than this. In that case, should we drop `WithAppendSlice`?
```suggestion
	return mergo.Merge(&c, config, mergo.WithOverride)
```

Reply:
In any case, I think we need tests exercising this logic. A testscript/txtar test at the root level might work well, since it is easy to stack multiple configs there

Reply:
i've added a txtar test.

i don't feel too strongly about WithAppendSlice, i initially only had WithOverride but it seemed to be a reasonable default for many slice config fields besides Node/Chain eg ListenAddresses and DefaultBootstrappers, although there probably aren't existing ones on the Node level. it also puts the responsibility on the plugin - it could make the decision to deduplicate, to override and keep only the last element etc. if slices were always overridden, no plugins would be aware of items in the slice in initial fragments without adding custom logic here. at the same time, it wouldn't be possible to know which items were specifically set in the last fragment if they did want to override.

Reply:
actually, just realizing that adding a Node in the 2nd fragment works because of WithAppendSlice, else we'd have to update RawConfig.SetFrom.

Reply:
We need to provide a consistent user experience though. Isn't the Node handling exceptional, just like Chains?

Reply:
i didn't want to add additional logic for matching Nodes - the fields might not be consistent the way we expect Chain Id to exist for the Chain struct - eg URL might not exist for a non-EVM chain where there are many different URLs. 

having WithAppendSlice to append all RawConfig slice fields seems fine to me - is your suggestion to only append for Node and to always override otherwise?

Reply:
I suggest that we drop `mergo.WithAppendSlice` and handle the one known case of Nodes directly, in a way that is consistent with the experience for every other chain. We have four `(*XYZNodes) SetFrom(*XYZNodes)`, and they all index on `Name`.

Reply:
updated, can you take a look?